hikari/impl/interaction_server.py

```diff
@@ -0,0 +1,514 @@
+            AIOHTTP's documentation.
+        """
+        if self._runner:
+            raise errors.ComponentStateConflictError("Cannot start an already active interaction server")
+
+        self._get_event().clear()
+        self._runner = aiohttp.web_runner.AppRunner(
+            self._server, handle_signals=enable_signal_handlers, access_log=_LOGGER
+        )
+        await self._runner.setup()
+        sites: typing.List[aiohttp.web.BaseSite] = []
+
+        if host is not None:
+            if isinstance(host, str):
+                host = [host]
+
+            for h in host:
+                sites.append(
+                    aiohttp.web.TCPSite(
+                        self._runner,
+                        h,
+                        port,
+                        shutdown_timeout=shutdown_timeout,
+                        ssl_context=ssl_context,
+                        backlog=backlog,
+                        reuse_address=reuse_address,
+                        reuse_port=reuse_port,
+                    )
+                )
+
+        elif path is None and socket is None or port is None:
+            sites.append(
+                aiohttp.web.TCPSite(
+                    self._runner,
+                    port=port,
+                    shutdown_timeout=shutdown_timeout,
+                    ssl_context=ssl_context,
+                    backlog=backlog,
+                    reuse_address=reuse_address,
+                    reuse_port=reuse_port,
+                )
+            )
+
+        if path is not None:
+            sites.append(
+                aiohttp.web.UnixSite(
+                    self._runner, path, shutdown_timeout=shutdown_timeout, ssl_context=ssl_context, backlog=backlog
+                )
+            )
+
+        if socket is not None:
+            sites.append(
+                aiohttp.web.SockSite(
+                    self._runner, socket, shutdown_timeout=shutdown_timeout, ssl_context=ssl_context, backlog=backlog
+                )
+            )
+
+        for site in sites:
+            _LOGGER.info("Starting site on %s", site.name)
+            await site.start()
+
+    def get_listener(
+        self, interaction_type: typing.Type[_InteractionT], /
+    ) -> typing.Optional[interaction_server.ListenerT[_InteractionT, special_endpoints.InteractionResponseBuilder]]:
+        return self._listeners.get(interaction_type)
+
+    def set_listener(
+        self,
+        interaction_type: typing.Type[_InteractionT],
+        listener: typing.Optional[
+            interaction_server.ListenerT[_InteractionT, special_endpoints.InteractionResponseBuilder]
+        ],
+        /,
+        *,
+        replace: bool = False,
+    ) -> None:
+        if listener:
+            if not replace and interaction_type in self._listeners:
+                raise TypeError(f"Listener already set for {interaction_type.__name__}")
+
+            self._listeners[interaction_type] = listener  # type: ignore[assignment]
```

Comment:
Why the type ignore?

Reply:
For the same reason the type ignores are in the event dispatcher, because that just ain't how the variance of callables works

Reply:
Should this be something we fix? I have seen a lof of these

Reply:
There's not really any way to "fix" it per-say, best you can do is just replace `typing.Callable[[Event], ...]` with `typing.Callable[[Any], ...]` in the typing for the listeners Dict to get rid of the type ignores